client/src/main/java/io/getstream/chat/android/client/helpers/AttachmentHelper.kt

```diff
@@ -0,0 +1,21 @@
+package io.getstream.chat.android.client.helpers
+
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.client.utils.SystemTimeProvider
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+
+class AttachmentHelper(private val systemTimeProvider: SystemTimeProvider = SystemTimeProvider()) {
+
+    fun hasValidUrl(attachment: Attachment): Boolean {
+        val url = attachment.url?.toHttpUrlOrNull() ?: return false
+        if (url.queryParameterNames.contains(QUERY_KEY_NAME_EXPIRES).not()) {
+            return true
+        }
+        return url.queryParameter(QUERY_KEY_NAME_EXPIRES)?.toLongOrNull()
+            ?.let { timestamp -> timestamp > systemTimeProvider.provideTime() } ?: false
```

Comment:
I believe this code is correct here, but it's the kind of suspicious structure with `let` and the Elvis operator that can occasionally lead to unexpected behaviour. It took me a while to think through all the possible cases of what can be `null` at what point, and what that causes.

How about going with something more conventional? (It's the same amount of code anyway, but much easier to verify what it does imo)

```suggestion
        val timestamp = url.queryParameter(QUERY_KEY_NAME_EXPIRES)?.toLongOrNull() ?: return false
        return timestamp > systemTimeProvider.provideTime()
```